rpc/src/error.rs

```diff
@@ -196,6 +209,7 @@ impl From<Code> for i32 {
         match code {
             Code::HttpError => 0,
             Code::WebSocketError => 1,
+            Code::ClientError => 2,
```

Comment:
Suggestion: this error should probably have a doc comment as it is "special" in the sense as it does not have a counterpart in golang.

Reply:
Yeah I was wondering whether or not to create a new error here. I see the `InternalError` code is reserved for RPC server internal errors, but needed to have something to represent an internal error specific to the client. I'd say it's better to differentiate between those two kinds of "internal" errors - what do you think @liamsi ?

Reply:
If we keep this then I may rename it to `ClientInternalError` and I'll add a doc comment indicating it doesn't have a counterpart in the Golang client.